sklearn/preprocessing/tests/test_data.py

```diff
@@ -604,6 +604,55 @@ def test_normalizer_l2():
         assert_almost_equal(la.norm(X_norm[3]), 0.0)
 
 
+def test_normalizer_max():
+    rng = np.random.RandomState(0)
+    X_dense = rng.randn(4, 5)
+    X_sparse_unpruned = sparse.csr_matrix(X_dense)
+
+    # set the row number 3 to zero
+    X_dense[3, :] = 0.0
+
+    # set the row number 3 to zero without pruning (can happen in real life)
+    indptr_3 = X_sparse_unpruned.indptr[3]
+    indptr_4 = X_sparse_unpruned.indptr[4]
+    X_sparse_unpruned.data[indptr_3:indptr_4] = 0.0
+
+    # build the pruned variant using the regular constructor
+    X_sparse_pruned = sparse.csr_matrix(X_dense)
+
+    # check inputs that support the no-copy optim
```

Comment:
I just copied the test structure from the `'l1'` and `'l2'` test cases, so if this is wrong, those are as well. I think it's checking the `copy=False` case correctly, though, when it asserts that `(X_norm2 is X)`